put-iam-role.yaml

```diff
@@ -0,0 +1,42 @@
+---
+platform: linux
+
+image_resource:
+  type: docker-image
+  source:
+    repository: governmentpaas/awscli
+
+inputs:
+  - name: common-tasks
+  - name: source
+
+outputs:
+- name: output
+
+params:
+  AWS_ACCESS_KEY_ID: ((role-putter-access-key-id))
+  AWS_SECRET_ACCESS_KEY: ((role-putter-secret-access-key))
+  AWS_NODES_ROLE_ARN: ((node-arn))
+  APP_NAME: ((app-name))
+
+run:
+  path: sh
+  args:
+  - -ecx
+  - |
+    sed 's@${AWS_NODES_ROLE_ARN}@'$AWS_NODES_ROLE_ARN'@g' ./common-tasks/templates/trust-policy.json.tmpl > ./trust-policy.json
+    export ROLE_NAME=$(cat source/deploy.json | jq -r '.role_name')
+    if [ -f "./source/iam/inline_policy.json" ]; then
+      if aws iam get-role --role-name $ROLE_NAME; then
+        echo "Updating Role ($ROLE_NAME) with Inline Policy:"
+      else
+        echo "Creating Role ($ROLE_NAME) with Inline Policy:"
+        aws iam create-role --role-name $ROLE_NAME --assume-role-policy-document file://./trust-policy.json
```

Comment:
When you'll get to the IAM policy for the role that creates these roles/policies watch out because the actions required to update inline policies are weirdly names.

I think to update inline policy you'll need to grant `iam:PutRolePolicy`, https://iam.cloudonaut.io/reference/iam.html is your friend 🏆